### PR TITLE
replaced dojo/topic w/ component outputs (events) and local variables

### PR DESCRIPTION
**Pull request by tomwayson**
not sure if I've gotten this exactly right (Angular 2 is new to me too), but it works, and seems like a more angular-ish way of having components communicate w/ each other.

### CODE REVIEW

File: app/home.component.ts [hunk 1]
```diff
@@ -9,21 +9,19 @@ export class HomeComponent {
...
+    this.vm = new HomeViewModel({});
   }
 
-
-  onMapView(view) {
-    this.vm = new HomeViewModel({
-      view: view
-    });
+  setView(view) {
+    this.vm.set('view', view);
```
Comment (odoe) [inline app/home.component.ts:17]: Can we just set this to `this.vm.view = view`? The ViewModels are accessors.

Comment (tomwayson) [inline app/home.component.ts:17]: yes, we can. I've tested it, and it works. All staged and ready to push. any other changes before I do?

**Comment by odoe** [comment]
Awesome, thanks Tom!

I had issues with EventEmitter and `next()` in my first pass. Not sure what I did wrong, but thanks for fixing it!

**Comment by tomwayson** [comment]
I did too - I tried to pass a reference to the map view via `.next()` - but when I tried to handle the event in the app component, the reference was null. That's when I switched to using local variables for the components (`#mapView` and `#homeButton`) w/ in the template. Seems to work fine, but not sure if it's the "right" way.

**Pull request merged**